staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go

```diff
@@ -598,6 +598,9 @@ const int64QuantityExpectedBytes = 18
 // String is an expensive operation and caching this result significantly reduces the cost of
 // normal parse / marshal operations on Quantity.
 func (q *Quantity) String() string {
+	if q == nil {
+		return "nil"
+	}
```

Comment:
It is convention to check for `nil` receiver as the caller rather than inside the method body. Returning a value like this has serious implications (more info [here](https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_never_use_nil_to_indicate_failure)).

Reply:
According to https://raw.githubusercontent.com/kubernetes/kubernetes/master/staging/src/k8s.io/api/core/v1/generated.pb.go

```
func (this *Pod) String() string {
	if this == nil {
		return "nil"
	}
	s := strings.Join([]string{`&Pod{`,
		`ObjectMeta:` + strings.Replace(strings.Replace(fmt.Sprintf("%v", this.ObjectMeta), "ObjectMeta", "v1.ObjectMeta", 1), `&`, ``, 1) + `,`,
		`Spec:` + strings.Replace(strings.Replace(this.Spec.String(), "PodSpec", "PodSpec", 1), `&`, ``, 1) + `,`,
		`Status:` + strings.Replace(strings.Replace(this.Status.String(), "PodStatus", "PodStatus", 1), `&`, ``, 1) + `,`,
		`}`,
	}, "")
	return s
}
```

The generated String function always check `nil` receiver. And many kubernetes structs contain *resource.Quantity field, the default value is nil. The String func always cause panic although it was catched by `fmt/print.go catchPanic`

Reply:
That is panic stacktrace
![image](https://user-images.githubusercontent.com/2086945/97301980-66452a00-1893-11eb-9dec-931705e6804b.png)